src/snowflake/snowpark/files.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+"""SnowflakeFile for UDFs in Snowpark."""
+from __future__ import annotations
+
+import array
+from typing import Iterable
+
+from snowflake.snowpark._internal.utils import private_preview
+
+_DEFER_IMPLEMENTATION_ERR_MSG = (
+    "The file operations is implemented in Snowflake servers"
+)
+
+
+class SnowflakeFile:
+    """
+    SnowflakeFile provides developers an interface to operate files as a Python IOBase-like object in UDFs.
+    SnowflakeFile supports most operations supported by Python IOBase
+    For more information about the class, please check ...(docs TBD)
```

Comment:
Before the doc link is ready, we may remove this sentence.

Reply:
Yes, please remove this text:
"For more information about the class, please check ...(docs TBD)"

Reply:
fixed